documentation/internal/Level-1-Review-And-LOQ.md

```diff
@@ -0,0 +1,120 @@
+
+## Review Responses Visibility
+
+`is_visible_to_applicant` field (not editable by any user, only by back end) controls visibility of review_response from applicant perspective. And `recommended_applicant_visibility` field (which is editable by reviewer) suggests visibility of review_response to applicant. `updateReviewVisibility` back end action will set `is_visible_to_applicant` when review is submitted. See `LOQ Workflow Diagram` below
+
+## Actions
+
+Front end determines user actions based on review and application status. And theses statuses are changed by back end via actions, here is an abstracted diagram:
+
+![Review-And-LOQ-Actions](images/Review-And-LOQ-Actions.png)
+
+Actions for review are computed by section via `useGetReviewStructureForSections` -> `generateReviewStructure` -> `generateReviewSectionActions`
+
+Actions for applications `changeRequestsProgress` and `isChangeRequest` is used to display actions in Application Home pages, those are generated in `useGetApplicationStructure` -> `addApplicantChangeRequestStatusToElement` and `generateApplicantChangesRequestedProgress`.
+
+See `Structure` below for more details
+
+## Validation of Submission
+
+When **application** is restarted as a result of LOQ, it can only be submitted if general validation passes AND all questions with change requests have been changed. Logic for this is:
+
+latestApplicationResponse is different to previousApplicationResponse where previousApplicationResponse is linked to non confirm reviewResponse
+
+**review** can be submitted as `LOQ` or `Non-conform` when at least review_response is marked as `decline`. Otherwise submission of `Conform` can be made if ALL assigned visible latest application responses have `approve` status review responses.
+
+One thing to note, since we always look at latest application response and it's linked review response, we guarantee that all new changes by applicant will be re-reviewed
+
+## Structure
+
+Application and Review Structure are generated in hooks `useGetApplicationStructure` and `useGetReviewStructureForSection` respectively, these structures are re-computed whenever responses are changed. In the context of Review and LOQ, these fields are added:
+
+### Application
+
+![Applicant LOQ Structure Changes](images/Applicant-LOQ-Structure-Changes.png)
+
+### Review
+
+![Level-1-Review-Structure-Changes](images/Level-1-Review-Structure-Changes.png)
+
+## Components and Hooks
+
+This is described in Application and Review Flow documentation
+
+# LOQ Workflow Diagram
+
+![LOQ Workflow](images/LOQ-Workflow.png)
+
+# Responses Workflow Diagram
+
+Note this diagram may be slightly outdated, but it is still indicative of implemented logic. It also goes beyond Review and LOQ topic
+
+![Responses Workflow](images/Responses-Workflow.png)
```

Comment:
Haven't done a review for this one. Will leave it for re-review, but have in mind it doesn't need to be exact up to date, just reflect what we have been doing currently.

Reply:
I think it's generally pretty accurate. It should give a good idea of the implementation. I wish there was an easier way to update diagram and even link them to code they relate to